src/backend/columnar/cstore_metadata_tables.c

```diff
@@ -498,10 +532,12 @@ GetHighestUsedAddressAndId(Oid relfilenode,
 	SnapshotData SnapshotDirty;
 	InitDirtySnapshot(SnapshotDirty);
 
-	List *stripeMetadataList = ReadDataFileStripeList(relfilenode, &SnapshotDirty);
+	List *stripeMetadataList = ReadDataFileStripeList(storageId, &SnapshotDirty);
 
 	*highestUsedId = 0;
-	*highestUsedAddress = 0;
+
+	/* file starts with metapage */
+	*highestUsedAddress = sizeof(ColumnarMetapage);
```

Comment:
Can we reserve the complete first postgres block of the buffer as a metapage? using the size of the struct as the highestUsedAddress we have little flexibility in increasing its size without completely invalidating the data already stored. If we reserve the first block we give ourself enough headroom to change the format in the future.

Reply:
agreed. done.